files/default/sql/ddl/updates/3.3.0.sql

```diff
@@ -100,3 +100,229 @@ ALTER TABLE `hopsworks`.`transformation_function` DROP KEY `inode_fn_fk`;
+                                           `row_percentage` FLOAT NOT NULL DEFAULT 1.0,
+                                           PRIMARY KEY (`id`),
+                                           KEY `feature_view_id` (`feature_view_id`),
+                                           KEY `window_start_event_time` (`window_start_event_time`),
+                                           KEY `window_end_event_time` (`window_end_event_time`),
+                                           CONSTRAINT `fvs_fv_fk` FOREIGN KEY (`feature_view_id`) REFERENCES `feature_view` (`id`) ON DELETE CASCADE ON UPDATE NO ACTION
+) ENGINE=ndbcluster DEFAULT CHARSET=latin1 COLLATE=latin1_general_cs;
+
+CREATE TABLE IF NOT EXISTS `training_dataset_statistics` (
+                                           `id` int(11) NOT NULL AUTO_INCREMENT,
+                                           `commit_time` DATETIME(3) NOT NULL,
+                                           `training_dataset_id`INT(11) NOT NULL,
+                                           `for_transformation` TINYINT(1) DEFAULT '0',
+                                           `row_percentage` FLOAT NOT NULL DEFAULT 1.0,
+                                           PRIMARY KEY (`id`),
+                                           KEY `training_dataset_id` (`training_dataset_id`),
+                                           CONSTRAINT `tds_td_fk` FOREIGN KEY (`training_dataset_id`) REFERENCES `training_dataset` (`id`) ON DELETE CASCADE ON UPDATE NO ACTION
+) ENGINE=ndbcluster DEFAULT CHARSET=latin1 COLLATE=latin1_general_cs;
+
+-- TODO: SQL Queries for populating new tables based on feature_store_statistic table.
+
+-- Update feature_store_activity foreign keys
+ALTER TABLE `hopsworks`.`feature_store_activity`
+    ADD COLUMN `feature_group_statistics_id` INT(11) NULL,
+    ADD COLUMN `feature_view_statistics_id` INT(11) NULL,
+    ADD COLUMN `training_dataset_statistics_id` INT(11) NULL,
+    ADD CONSTRAINT `fs_act_fg_stat_fk` FOREIGN KEY (`feature_group_statistics_id`) REFERENCES `feature_group_statistics` (`id`) ON DELETE CASCADE ON UPDATE NO ACTION,
+    ADD CONSTRAINT `fs_act_fv_stat_fk` FOREIGN KEY (`feature_view_statistics_id`) REFERENCES `feature_view_statistics` (`id`) ON DELETE CASCADE ON UPDATE NO ACTION,
+    ADD CONSTRAINT `fs_act_td_stat_fk` FOREIGN KEY (`training_dataset_statistics_id`) REFERENCES `training_dataset_statistics` (`id`) ON DELETE CASCADE ON UPDATE NO ACTION;
+-- TODO: SQL Query populate ids based on statistics_id.
+ALTER TABLE `hopsworks`.`feature_store_activity`
+    DROP FOREIGN KEY `fs_act_stat_fk`,
+    DROP COLUMN `statistics_id`;
+
+ALTER TABLE `hopsworks`.`feature_store_statistic`
+    DROP FOREIGN KEY `fg_fk_fss`,
+    DROP FOREIGN KEY `fg_ci_fk_fss`,
+    DROP FOREIGN KEY `td_fk_fss`,
+    DROP FOREIGN KEY `inode_fk`;
+DROP TABLE IF EXISTS `hopsworks`.`feature_store_statistic`;
```

Comment:
Why do we need to drop the FK if we are just dropping the entire table?

Reply:
Additionally: 
- How do we migrate old statistics to the new format? 
- How do we handle feature group correlation and histograms?

Reply:
> Why do we need to drop the FK if we are just dropping the entire table?

Fixed.

> How do we migrate old statistics to the new format?

The idea is to implement an expat migration that reads the hdfs files, extracts and stores the descriptive statistics into the database, and if histogram or correlations are available, updates the hdfs file accordingly and stores the string path in the db.

For the feature store activity table, just SQL queries should work.

> How do we handle feature group correlation and histograms?

They are kept in hdfs files, which path is stored in a column in the feature descriptive statistics table.